x-pack/plugins/secops/public/components/range_date_picker/index.tsx

```diff
@@ -0,0 +1,201 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { EuiDatePickerRange, EuiFlexGroup, EuiFlexItem, EuiFormControlLayout } from '@elastic/eui';
+import { get, getOr } from 'lodash/fp';
+import moment, { Moment } from 'moment';
+import React from 'react';
+import { connect } from 'react-redux';
+import { inputsActions, inputsModel, State } from '../../store';
+import { GlobalDateButton } from './global_date_button';
+import { QuickSelectPopover } from './quick_select_popover';
+import { UpdateButton } from './update_button';
+
+export type DateType = 'relative' | 'absolute';
+
+interface RangeDatePickerStateRedux {
+  from: number;
+  to: number;
+  isTimerOn: boolean;
+  duration: number;
+  isLoading: boolean;
+  refetch: inputsModel.Refetch[];
+}
+
+interface RangeDatePickerDispatchProps {
+  setAbsoluteRangeDatePicker: (params: { id: string; from: number; to: number }) => void;
+  setRelativeRangeDatePicker: (
+    params: { id: string; option: string; from: number; to: number }
+  ) => void;
+  startAutoReload: (params: { id: string }) => void;
+  stopAutoReload: (params: { id: string }) => void;
+  setDuration: (params: { id: string; duration: number }) => void;
+}
+interface OwnProps {
+  id: string;
+  disabled?: boolean;
+}
+
+type RangeDatePickerProps = OwnProps & RangeDatePickerDispatchProps & RangeDatePickerStateRedux;
+
+export interface RecentlyUsedI {
+  kind: string;
+  text: string | number[];
+}
+
+interface RangeDatePickerState {
+  recentlyUsed: RecentlyUsedI[];
+}
+
+class RangeDatePickerComponents extends React.PureComponent<
+  RangeDatePickerProps,
+  RangeDatePickerState
+> {
+  public readonly state = {
+    recentlyUsed: [] as RecentlyUsedI[],
+  };
+
+  public render() {
+    const { recentlyUsed } = this.state;
+    const { id, isLoading, disabled = false, from, to, isTimerOn, refetch } = this.props;
+
+    const quickSelectPopover = (
+      <QuickSelectPopover
+        disabled={disabled}
+        recentlyUsed={recentlyUsed}
+        isTimerOn={isTimerOn}
+        onChange={this.onChange}
+        updateAutoReload={this.updateTimer}
+      />
+    );
+
+    return (
+      <EuiFlexGroup gutterSize="s" responsive={false}>
+        <EuiFlexItem style={{ maxWidth: 480 }}>
+          <EuiFormControlLayout className="euiGlobalDatePicker" prepend={quickSelectPopover}>
+            <EuiDatePickerRange
+              className="euiDatePickerRange--inGroup"
+              iconType={false}
+              isCustom
+              startDateControl={
+                <GlobalDateButton
+                  id={`${id}From`}
```

Comment:
I would use a delimiter of `.` here as in other areas of the code we use it when doing id tricks.